app/components/channel_list/search/index.tsx

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {StyleSheet, View} from 'react-native';
+import {TextInput} from 'react-native-gesture-handler';
+
+import {useTheme} from '@app/context/theme';
+import {typography} from '@app/utils/typography';
+import CompassIcon from '@components/compass_icon';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => ({
+    container: {
+        flex: 1,
+        flexDirection: 'row',
+        justifyContent: 'flex-start',
+        alignItems: 'center',
+        width: '100%',
+        backgroundColor: changeOpacity(theme.sidebarText, 0.12),
+        borderRadius: 8,
+        padding: 8,
+        marginVertical: 20,
+        height: 40,
+        maxHeight: 40,
+    },
+    icon: {
+        width: 24,
+        fontSize: 24,
+        color: changeOpacity(theme.sidebarText, 0.72),
+    },
+    input: {
+        flex: 1,
+        height: 40,
+        color: theme.sidebarText,
+        alignItems: 'center',
+        alignContent: 'center',
+        marginLeft: 5,
+        marginTop: -2,
+    },
+}));
+
+const textStyles = StyleSheet.create([
+    typography('Body', 200),
+    {textAlignVertical: 'center'},
+]);
+
+const SearchField = () => {
+    const theme = useTheme();
+    const styles = getStyleSheet(theme);
+
+    return (
+        <View
+            style={styles.container}
+        >
+            <CompassIcon
+                name='magnify'
+                style={styles.icon}
+            />
+            <TextInput
```

Comment:
Ok for now, we don't have anyone really working on the SearchBar component and the one from v1 is sort of a bad one.. so perhaps let's create a p0 ticket for that.

FYI @Willyfrog

Reply:
[ticket created](https://mattermost.atlassian.net/browse/MM-39872)